src/Microsoft.Extensions.Hosting.Abstractions/IApplicationEnvironment.cs

```diff
@@ -0,0 +1,23 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+namespace Microsoft.Extensions.Hosting
+{
+    /// <summary>
+    /// Provides information about the hosting environment an application is running in.
+    /// </summary>
+    public interface IApplicationEnvironment
+    {
+        /// <summary>
+        /// Gets or sets the name of the environment. This property is automatically set by the host to the value
+        /// of the "DOTNETCORE_ENVIRONMENT" environment variable.
+        /// </summary>
+        string EnvironmentName { get; set; }
+
+        /// <summary>
+        /// Gets or sets the name of the application. This property is automatically set by the host to the assembly containing
+        /// the application entry point.
+        /// </summary>
+        string ApplicationName { get; set; }
+    }
```

Comment:
You can set them on the ConfigurationBuilder if you need the base path for that. So config on its own isn't enough reason to add it. I am not convinced we should add it unless we have some scenarios in mind where you would want ContentRootPath instead of the alternatives. Do you have anything specific or just looking for gaps?